test/explain/algoritms/test_gnn_explainer.py

```diff
@@ -0,0 +1,672 @@
+    model,
+    return_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type="model",
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='node',
+        return_type=return_type,
+        mode="classification",
+    )
+
+    explainer = GNNExplainer()
+    model = model()
+
+    x = torch.randn(8, 3)
+    edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7],
+                               [1, 0, 2, 1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6]])
+    out = model(x, edge_index).argmax(dim=-1)
+
+    # try to explain prediction for node 0
+    explanation = explainer(x=x, edge_index=edge_index, model=model,
+                            target=out, target_index=0,
+                            explainer_config=explainer_config,
+                            model_config=model_config, node_index=2)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('model', [
+    GNN_classification_multioutput,
+    GNN_classification_singleoutput,
+])
+@pytest.mark.parametrize('return_type', return_types_classification)
+def test_gnn_explainer_graph_classification(
+    edge_mask_type,
+    node_mask_type,
+    model,
+    return_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type="model",
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='graph',
+        return_type=return_type,
+        mode="classification",
+    )
+
+    explainer = GNNExplainer()
+    model = model()
+
+    x = torch.randn(8, 3)
+    edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7],
+                               [1, 0, 2, 1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6]])
+    out = model(x, edge_index, None, None).argmax(dim=-1)
+
+    if isinstance(model, GNN_classification_multioutput):
+        target_index = 0
+    else:
+        target_index = None
+
+    # try to explain prediction for node 0
+    explanation = explainer(x=x, edge_index=edge_index, model=model,
+                            target=out, target_index=target_index,
+                            explainer_config=explainer_config,
+                            model_config=model_config, edge_attr=None,
+                            batch=None)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('model', [
+    GNN_regression_singleoutput,
+    GNN_regression_multioutput,
+])
+@pytest.mark.parametrize('return_type', return_types_regression)
+def test_gnn_explainer_graph_regression(
```

Comment:
Correct me if I am wrong but doesn't `test_gnn_explainer_with_meta_explainer_regression_graph` make this test redundant?

Reply:
yes, they were just useful when rewriting the algorithm, now I will delete them